sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/_tests_/e2e/ier12-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,394 @@
+import { DeepMocked } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames } from "@sims/utilities";
+import * as dateUtils from "@sims/utilities/date-utils";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../../../test/helpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  getUploadedFile,
+  saveFakeStudentRestriction,
+} from "@sims/test-utils";
+import * as Client from "ssh2-sftp-client";
+import { ArrayContains, Not } from "typeorm";
+import {
+  ApplicationStatus,
+  AssessmentTriggerType,
+  COEStatus,
+  DisbursementScheduleStatus,
+  RelationshipStatus,
+  RestrictionActionType,
+} from "@sims/sims-db";
+import { IER12IntegrationScheduler } from "../../ier12-integration.scheduler";
+import { saveIER12TestInputData } from "./ier12-factory";
+import {
+  ASSESSMENT_DATA_MARRIED,
+  ASSESSMENT_DATA_SINGLE_DEPENDENT,
+  ASSESSMENT_DATA_SINGLE_INDEPENDENT,
+  AWARDS_ONE_OF_TWO_DISBURSEMENT,
+  AWARDS_SINGLE_DISBURSEMENT,
+  AWARDS_SINGLE_DISBURSEMENT_NO_BC_FUNDING,
+  AWARDS_TWO_OF_TWO_DISBURSEMENT,
+  JANE_MONONYMOUS_FROM_OTHER_COUNTRY,
+  JOHN_DOE_FROM_CANADA,
+  OFFERING_FULL_TIME,
+  PROGRAM_GRADUATE_DIPLOMA_WITH_INSTITUTION_PROGRAM_CODE,
+  PROGRAM_UNDERGRADUATE_CERTIFICATE_WITHOUT_INSTITUTION_PROGRAM_CODE,
+  WORKFLOW_DATA_DEPENDANT_RELATIONSHIP_OTHER_LIVING_WITH_PARENTS,
+  WORKFLOW_DATA_MARRIED_WITH_DEPENDENTS,
+  WORKFLOW_DATA_SINGLE_INDEPENDENT_WITH_NO_DEPENDENTS,
+} from "./models/data-inputs";
+import { QueueProcessSummaryResult } from "../../../../../../processors/models/processors.models";
+import { numberToText, getSuccessSummaryMessages } from "./utils/string-utils";
+import { createIER12SchedulerJobMock, isValidFileTimestamp } from "./utils";
+
+describe(describeProcessorRootTest(QueueNames.IER12Integration), () => {
+  let app: INestApplication;
+  let processor: IER12IntegrationScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+  let sharedProgramYearPrefix: number;
+  let referenceSubmissionDate: Date;
+  let getFileNameAsCurrentTimestampMock: jest.SpyInstance;
+  const defaultApplicationNumber = "9900000001";
+
+  beforeAll(async () => {
+    process.env.INSTITUTION_REQUEST_FOLDER = "Institution-Request";
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(IER12IntegrationScheduler);
+    // Default program year prefix.
+    sharedProgramYearPrefix = 2000;
+    referenceSubmissionDate = new Date("2000-06-01");
```

Comment:
Why this isn't a const as it is not re-assigned anywhere else?

Reply:
No particular reason, I will declare in the same way that `defaultApplicationNumber`.

Reply:
Changed.